Sources/ArcGISToolkit/Components/OverviewMap.swift

```diff
@@ -43,37 +41,57 @@ public struct OverviewMap: View {
     /// - Parameters:
     ///   - viewpoint: Viewpoint of the main `MapView` used to update the `OverviewMap` view.
     ///   - visibleArea: Visible area of the main `MapView ` used to display the extent graphic.
+    ///   - map: The `Map` displayed in the `OverviewMap`. Defaults to `nil`, in which case
+    ///   a map with the `arcGISTopographic` basemap style is used.
     /// - Returns: A new `OverviewMap`.
     public static func forMapView(
         with viewpoint: Viewpoint?,
-        visibleArea: ArcGIS.Polygon?
+        visibleArea: ArcGIS.Polygon?,
+        map: Map? = nil
     ) -> OverviewMap {
-        OverviewMap(viewpoint: viewpoint, visibleArea: visibleArea, symbol: .defaultFill)
+        OverviewMap(
+            viewpoint: viewpoint,
+            visibleArea: visibleArea,
+            symbol: .defaultFill,
+            map: map
+        )
     }
-    
+
     /// Creates an `OverviewMap` for use on a `SceneView`.
-    /// - Parameter viewpoint: Viewpoint of the main `SceneView` used to update the
+    /// - Parameters:
+    ///   - viewpoint: Viewpoint of the main `SceneView` used to update the
     /// `OverviewMap` view.
+    ///   - map: The `Map` displayed in the `OverviewMap`. Defaults to `nil`, in which case
+    ///   a map with the `arcGISTopographic` basemap style is used.
     /// - Returns: A new `OverviewMap`.
     public static func forSceneView(
-        with viewpoint: Viewpoint?
+        with viewpoint: Viewpoint?,
+        map: Map? = nil
     ) -> OverviewMap {
-        OverviewMap(viewpoint: viewpoint, symbol: .defaultMarker)
+        OverviewMap(viewpoint: viewpoint, symbol: .defaultMarker, map: map)
     }
     
     /// Creates an `OverviewMap`. Used for creating an `OverviewMap` for use on a `MapView`.
     /// - Parameters:
     ///   - viewpoint: Viewpoint of the main `GeoView` used to update the `OverviewMap` view.
     ///   - visibleArea: Visible area of the main `GeoView` used to display the extent graphic.
+    ///   - map: The `Map` displayed in the `OverviewMap`.
     init(
         viewpoint: Viewpoint?,
         visibleArea: ArcGIS.Polygon? = nil,
-        symbol: Symbol
+        symbol: Symbol,
+        map: Map?
     ) {
         self.visibleArea = visibleArea
         self.viewpoint = viewpoint
         self.symbol = symbol
         
+        _dataModel = StateObject(
+            wrappedValue: MapDataModel(
+                map: map ?? Map(basemapStyle: .arcGISTopographic)
+            )
+        )
```

Comment:
I don't know that this will do what you'd expect. Say the user has two maps, one light and one dark, and they toggle between the two. A new state object is created each time, which isn't how state objects are intended to be used. Instead, I would suggest storing the user provided map on the view and creating the default map in the view model. That way the view model itself, and therefore the default map, will only be created if necessary (remember that `StateObject.init(wrappedValue:)` takes an autoclosure, so the wrapped value is lazily evaluated the first time the state object is accessed). It would look something like this:

```swift
// In 'OverviewMap'
private class Model: ObservableObject {
    let defaultMap = Map(basemapStyle: .arcGISTopographic)
}

@StateObject private var model = Model()

private let userProvidedMap: Map?

private var effectiveMap: Map {
    userProvidedMap ?? model.defaultMap
}

// In the 'body' of 'OverviewMap'

MapView(
    map: effectiveMap,
    viewpoint: makeOverviewViewpoint(),
    graphicsOverlays: [graphicsOverlay]
)
```